utility.py

```diff
@@ -0,0 +1,139 @@
+from concurrent import futures
+from azure.storage.blob import BlockBlobService, PublicAccess
+import os,uuid,sys
+import io
+import grpc
+import time
+import os.path
+import re
+import config
+import hwsc_file_transaction_svc_pb2
+import hwsc_file_transaction_svc_pb2_grpc
+
+CHUNK_SIZE = 1024 * 1024
+block_blob_service = BlockBlobService(account_name=config.CONFIG["storage"],
+                                          account_key=config.CONFIG["storage_key"])
+
+def download_chunk(file):
+     """Download a file in chunks."""
+     with open(file, "rb")as f:
+         while True:
+             chunk = f.read(CHUNK_SIZE)
+             if len(chunk) == 0:
+                 return
+             yield hwsc_file_transaction_svc_pb2.chunk(buffer=chunk)
+
+def get_file_type(file_name):
+    """Check and return correct type of the file passed in."""
+    image_exts_dict = {"jpg": True, "jpeg": True, "png": True, "bmp": True, "tif": True, "gif": True, "tiff": True}
+    audio_exts_dict = {"wav": True, "wma": True, "ogg": True, "m4a": True, "mp3": True}
+    video_exts_dict = {"flv": True, "wmv": True, "mov": True, "avi": True, "mp4": True}
+
+    file_list = file_name.split('.')
+    extension = file_list[-1]
+    file_type = "files"
+
+    if image_exts_dict.get(extension):
+        file_type = "images"
+    elif audio_exts_dict.get(extension):
+        file_type = "audios"
+    elif video_exts_dict.get(extension):
+        file_type = "videos"
+
+    return file_type
+
+def upload_file_to_azure(stream, has_folder, uuid, file_name):
+    """Upload file to azure blob storage."""
+    #TODO
+    #ADD try-catch block
+
+    # Checking whether the blobs that associated with uuid exists
+    if has_folder:
+        container_type = get_file_type(file_name)
+        container_name = uuid + '-' + container_type
+
+        # Set the permission so the blobs are public.
+        block_blob_service.set_container_acl(container_name, public_access=PublicAccess.Container)
```

Comment:
what happens if this fails? i'm assuming the try catch block will catch it? (i know its todo)